### PR TITLE
Unicode, String internals

### PR DESCRIPTION
**Pull request by mostafa-nematpour**
translate `Unicode, String internals`

### CODE REVIEW

File: 1-js/99-js-misc/06-unicode/article.md [hunk 1]
```diff
@@ -1,172 +1,175 @@
 
-# Unicode, String internals
+# یونی‌کد، رشته های داخلی
```
Comment (mahdyar) [inline 1-js/99-js-misc/06-unicode/article.md:2]:
```suggestion
# یونی‌کد، درون رشته‌ها
```

File: 1-js/99-js-misc/06-unicode/article.md [hunk 2]
```diff
@@ -1,172 +1,175 @@
 
-# Unicode, String internals
+# یونی‌کد، رشته های داخلی 
 
-```warn header="Advanced knowledge"
-The section goes deeper into string internals. This knowledge will be useful for you if you plan to deal with emoji, rare mathematical or hieroglyphic characters, or other rare symbols.
+```warn header="اطلاعات بیشتر"
+این بخش به دنبال عمیق شدن در رشته‌ها است. این دانش برای شما مفید خواهد بود اگر قصد دارید با شکلک‌ها، کاراکترهای ریاضی نادر یا هیروگلیف‌ها یا سایر نمادهای نادر سر و کار داشته باشید.
```
Comment (mahdyar) [inline 1-js/99-js-misc/06-unicode/article.md:5]:
```suggestion
این بخش به دنبال عمیق شدن در رشته‌ها است. این دانش برای شما مفید خواهد بود اگر قصد دارید با ایموجی‌ها، کاراکترهای ریاضی نادر یا هیروگلیف‌ها یا سایر نمادهای نادر سر و کار داشته باشید.
```

**Review by mahdyar — CHANGES_REQUESTED** [review]

File: 1-js/99-js-misc/06-unicode/article.md [hunk 3]
```diff
@@ -1,172 +1,175 @@
...
-    `XX` must be two hexadecimal digits with a value between `00` and `FF`, then `\xXX` is the character whose Unicode code is `XX`.
+    به جای `XX` باید دو عدد هگزادسیمال در بازه `00` و `FF` قرار گیرد. در نتیجه `‎\xXX` کاراکتری است که یونی‌کد آن `XX` هست.
 
-    Because the `\xXX` notation supports only two hexadecimal digits, it can be used only for the first 256 Unicode characters.
+    از آنجا که نماد `‎\xXX` فقط از دو عدد هگزادسیمال پشتیبانی می‌کند، از آن می‌توان فقط برای 256 کاراکتر اول یونی‌کد استفاده کرد.
+    
 
-    These first 256 characters include the Latin alphabet, most basic syntax characters, and some others. For example, `"\x7A"` is the same as `"z"` (Unicode `U+007A`).
+
+    این 256 کاراکتر اول شامل الفبای لاتین ، کاراکتر های نحوی ابتدایی و برخی چیزهای دیگر می‌شود. به عنوان مثال `"‎\x7A"` مانند نوشتن `"z"` هست. (یونی‌کد `U+007A`)
```
Comment (mahdiHash) [inline 1-js/99-js-misc/06-unicode/article.md:20]:
```suggestion
    این 256 کاراکتر اول شامل الفبای لاتین ، کاراکتر های نحوی ابتدایی و برخی چیزهای دیگر می‌شود. به عنوان مثال `"‎\x7A"` مانند نوشتن `"z"` هست. (یونی‌کد `U+007A`)
```
Please don't add an empty line.

File: 1-js/99-js-misc/06-unicode/article.md [hunk 4]
```diff
@@ -1,172 +1,175 @@
...
     alert( "\x7A" ); // z
-    alert( "\xA9" ); // ©, the copyright symbol
+    alert( "\xA9" ); // ©, نماد کپی‌رایت
     ```
 
-- `\uXXXX`
-    `XXXX` must be exactly 4 hex digits with the value between `0000` and `FFFF`, then `\uXXXX` is the character whose Unicode code is `XXXX`.
+- `‎\uXXXX`
+  
+    به جای `XXXX` باید 4 عدد هگزادسیمال در بازه `0000` و `FFFF` قرار گیرد. در نتیجه `‎\uXXXX` کاراکتری است که یونی‌کد آن `XXXX` هست.
```
Comment (mahdiHash) [inline 1-js/99-js-misc/06-unicode/article.md:29]:
```suggestion
 
    به جای `XXXX` باید 4 عدد هگزادسیمال در بازه `0000` و `FFFF` قرار گیرد. در نتیجه `‎\uXXXX` کاراکتری است که یونی‌کد آن `XXXX` هست.
```

File: 1-js/99-js-misc/06-unicode/article.md [hunk 5]
```diff
@@ -1,172 +1,175 @@
...
-    alert( "\u{1F60D}" ); // 😍, a smiling face symbol (another long Unicode)
+    alert( "\u{20331}" ); // 佫, یک کاراکتر چینی نادر (دارای یونی‌کد طولانی)
+    alert( "\u{1F60D}" ); // 😍, ایموجی چهره خندان (یونی‌کد طولانی دیگر)
     ```
 
-## Surrogate pairs
-
-All frequently used characters have 2-byte codes (4 hex digits). Letters in most European languages, numbers, and the basic unified CJK ideographic sets (CJK -- from Chinese, Japanese, and Korean writing systems), have a 2-byte representation.
+## جفت جایگزین (Surrogate pairs)
+تمامی کاراکترهای متداول دارای کدهای 2 بایتی هستند (4 رقم هگز). حروف در بیشتر زبان های اروپایی ، اعداد و مجموعه های ایدئوگرافیک چینی، ژاپنی و کره‌ای (CJK) ، با 2 بایت نمایش داده می‌شوند.
```
Comment (mahdiHash) [inline 1-js/99-js-misc/06-unicode/article.md:49]:
```suggestion
## جفت جایگزین (Surrogate pairs)

تمامی کاراکترهای متداول دارای کدهای 2 بایتی هستند (4 رقم هگز). حروف در بیشتر زبان های اروپایی ، اعداد و مجموعه های ایدئوگرافیک چینی، ژاپنی و کره‌ای (CJK) ، با 2 بایت نمایش داده می‌شوند.
```
Please don't remove any line.

File: 1-js/99-js-misc/06-unicode/article.md [hunk 6]
```diff
@@ -1,172 +1,175 @@
...
-They are essentially the same as [String.fromCharCode](mdn:js/String/fromCharCode) and [str.charCodeAt](mdn:js/String/charCodeAt), but they treat surrogate pairs correctly.
+این متدها بسیار شبیه [String.fromCharCode](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/fromCharCode) و [str.charCodeAt](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/fromCodePoint) هستند، اما در رفتار با جفت‌های جایگزین (surrogate pairs) به درستی عمل می‌کنند.
 
-One can see the difference here:
+تفاوت را در اینجا مشاهده کنید:
 
 ```js run
-// charCodeAt is not surrogate-pair aware, so it gives codes for the 1st part of 𝒳:
-
+// را می‌دهد 𝒳 به جفت‌ جایگزین آگاه نیست برای همین فقط کد قسمت اول charCodeAt
```
Comment (mahdiHash) [inline 1-js/99-js-misc/06-unicode/article.md:87]:
```suggestion
// را می‌دهد 𝒳 به جفت‌ جایگزین آگاه نیست برای همین فقط کد قسمت اول charCodeAt 

```

File: 1-js/99-js-misc/06-unicode/article.md [hunk 7]
```diff
@@ -1,172 +1,175 @@
...
+در بسیاری از زبان های نوشتاری، نمادهایی وجود دارند که از یک کاراکتر پایه با یک علامت در پایین یا بالای آن تشکیل شده‌اند.
 
-In many languages, there are symbols that are composed of the base character with a mark above/under it.
+به عنوان مثال حرف پایه می‌تواند `a` باشد و نمونه های مقابل را درست کرد: `àáâäãåā`
 
-For instance, the letter `a` can be the base character for these characters: `àáâäãåā`.
+در زبان فارسی هم نمونه‌هایی و جود دارد مانند تشدید در `ملّت` و حروف صدادار.
 
-Most common "composite" characters have their own code in the Unicode table. But not all of them, because there are too many possible combinations.
+اکثر کاراکتر های رایج ترکیبات خود را هم در جدول یونی‌کد دارند. ولی نه همه آنها، زیرا تعداد ترکیبات ممکن بسیار زیاد خواهد شد.
```
Comment (mahdiHash) [inline 1-js/99-js-misc/06-unicode/article.md:124]:
```suggestion
به عنوان مثال حرف پایه می‌تواند `a` باشد و نمونه های مقابل را درست کرد: `àáâäãåā`. در زبان فارسی هم نمونه‌هایی و جود دارد مانند تشدید در `ملّت` و حروف صدادار.

اکثر کاراکتر های رایج ترکیبات خود را هم در جدول یونی‌کد دارند. ولی نه همه آنها، زیرا تعداد ترکیبات ممکن بسیار زیاد خواهد شد.
```

**Review by mahdiHash — CHANGES_REQUESTED** [review]
Please review the suggestions.
You can add the suggestions to a batch and create a commit just once ([docs](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request#applying-suggested-changes)).

**Comment by mostafa-nematpour** [comment]
Thanks for the suggestions Mehdi
I will try to be better and next time I will put all in one
I'm not good at it
/done

**Comment by mahdiHash** [comment]
> Thanks for the suggestions Mehdi I will try to be better and next time I will put all in one I'm not good at it /done

No worries, mate :)
Thanks for your contribution, @mostafa-nematpour.

**Review by mahdiHash — APPROVED** [review]

**Pull request merged**